float32 iCoeff
uint8 period
